src/main/java/teammates/common/util/TimeHelper.java

```diff
@@ -330,6 +330,23 @@ public static boolean isOlderThanAYear(Date compareDate) {
     }
     
     /**
+     * Checks if the time falls within past hour from now.
+     * @param time the time to be checked
+     * @return true if the time falls within past hour from now.
+     */
+    public static boolean isWithinPastHour(Date time) {
+        Calendar now = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+
+        Calendar cal = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+        cal.setTime(time);
+
+        long nowMillis = now.getTimeInMillis();
+        long calMillis = cal.getTimeInMillis();
+        long differenceBetweenNowAndCal = (calMillis - nowMillis) / (60 * 60 * 1000);
+        return differenceBetweenNowAndCal == 0 && now.after(cal);
+    }
```

Comment:
The code can be like this:
```java
public static boolean isWithinPastHour(Date time) {
    return isWithinHour(time, getCurrentTime());
}
```
The  method `isWithinHour(t1,t2)` does not depend on the current time,  and therefore more testable. That is the one you should unit test thoroughly. For example, `isWithinPastHour(t1,t1)` should return false. I don't believe it is possible to test that scenario (where the given time matches the current time exactly) in the current code.